docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -4,14 +4,36 @@ description: Define tasks that can be run pre- or post-deployment
 weight: 89
 ---
 
-
 A `KeptnTaskDefinition` defines tasks
 that are run by the Keptn Lifecycle Toolkit
 as part of the pre- and post-deployment phases of a
 [KeptnApp](./app.md) or
 [KeptnWorkload](../concepts/workloads/).
 
-## Yaml Synopsis
+A Keptn task runs as an application
+[container](https://kubernetes.io/docs/concepts/containers/),
+which runs as part of a Keptn
+[job](https://kubernetes.io/docs/concepts/workloads/controllers/job/).
+A Keptn job is part of a task.
+
+TODO: That last sentence doesn't make sense --
+perhaps "task" needs a modifier.
+
+A Keptn task can be defined in either of two ways:
+
+* Define a Kubernetes application container,
+  that includes a runtime,  an application
+  and its runtime dependencies.
+  This gives you a lot of flexibility,
+  to define tasks using Python or Java scripts
+  or any anything else.
+* Out of the box, KLT includes a Deno-runtime container
```

Comment:
should we mention also a python runtime here? Or in a followup?

Reply:
Yes, I am sure we should.  I couldn't figure out where the source would be for this last week -- I tried looking at some PRs but didn't find it.

Reply:
I will start a pr collecting the info and maybe we can merge them 👍

Reply:
Thanks for the PR!  I just stole shamelessly from that PR to get coverage of python into the ref page.  It needs more work but now we have some coverage of python at least.